server/oauth.go

```diff
@@ -67,24 +67,15 @@ func (s *Server) registerOAuthRoutes(g *echo.Group) {
 			}
 		}
 
-		// We need to attach the RedirectURL in the get token process of oauth,
-		// and the RedirectURL needs to be consistent with the RedirectURL in the get code process.
-		// The frontend get it through window.location.origin in the get code process,
-		// so port 80 needs to be cropped when the backend splices the RedirectURL.
-		if s.profile.FrontendPort == 80 {
-			oauthExchange.RedirectURL = fmt.Sprintf("%s/oauth/callback", s.profile.FrontendHost)
-		} else {
-			oauthExchange.RedirectURL = fmt.Sprintf("%s:%d/oauth/callback", s.profile.FrontendHost, s.profile.FrontendPort)
-		}
-
+		oauthExchange.RedirectURL = fmt.Sprintf("%s/oauth/callback", s.profile.ExternalURL)
```

Comment:
Do we need to normalize ExternalURL so that it does not contains a trailing slash?

Reply:
start() will do this. https://github.com/bytebase/bytebase/pull/2608/files#diff-ebc8310b48b188c4f24d8e33a17aff4e4c88226502d2bdf69000a6ba7eed94d9R240-R242

Reply:
Right, normalized here

https://github.com/bytebase/bytebase/pull/2608/files#diff-3138996b8c0a5a8f6d8de8f35c69f14a46eef9d2a44f1844fced269922546b5fR39